staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -168,6 +168,10 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr admission.Attrib
 		if err != nil {
 			switch err := err.(type) {
 			case *webhookutil.ErrCallingWebhook:
+				if ctx.Err() == context.Canceled {
+					klog.Warningf("Context Canceled when calling webhook %v", hook.Name)
+					return err
```

Comment:
A few thoughts here:

1. Previously, this block was only about recording metrics... if we can limit the returns to the last block in the loop, it will make it easier not to lose track of the control flow.

2. If the webhook taking too long was *why* the context timed out, shouldn't we still record that in metrics somehow? This looks like we lose all insights into webhook latency if the overall request times out.

3. Returning the raw error here is a behavior change... shouldn't we return an API error (either `return apierrors.NewTimeoutError("request did not complete within requested timeout", 0)` or `apierrors.NewInternalError(err)`) as we did previously?

Reply:
2. when the webhook actually times out we've been seeing it as `context deadline exceeded` rather than `context cancelled`
3. Yeah @divyasri537 we should make it return the `apierrors.NewInternalError(err)` but I don't think a client would care what is returned? since they already cancelled their request? But yeah still should be `apierrors.NewInternalError(err)`